### PR TITLE
First commit for Master ID implementation

### PR DESCRIPTION
**Pull request by Vlad-CL**
# JIRA

https://clxmedia.atlassian.net/browse/CLXD-1689

# Description

Naming the table for describing connections between the Odoo and platforms as **platform.connector**
List of all platforms storing in **platform.type**

### CODE REVIEW

File: contact_modification/models/platform_connection.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    template_string = template_string.replace("{adgroup}", rec.ad_group_id)
+                rec.href = template_string
+            else:
+                rec.href = False
+
+    def edit_form(self):
+        return {
+            "type": "ir.actions.act_window",
+            "name": "Edit Connection",
+            # "view_type": "tree",
```
Comment (dmontooth) [inline contact_modification/models/platform_connection.py:87]: If we don't need these commented lines, remove please.

File: contact_modification/models/res_partner.py [hunk 2]
```diff
@@ -213,6 +214,27 @@ def _compute_task_count(self):
     tag_manager_place = fields.Many2one("res.partner.tagmanagerplacement", string="Tag Manager Placement", default=1)
     tag_manager_id = fields.Char(string="Google Tag Manager ID", help="Format: GTM-XXXXXX")
 
+    platform_connector_ids = fields.One2many("platform.connector", "partner_id", string="Platform Connector")
+
+    def action_platform_connection_wizard(self):
+        view_id = self.env.ref("contact_modification.platform_connection_form").id
+        platform_type_object = self.env["platform.type"]
+        # account_field_name =
```
Comment (dmontooth) [inline contact_modification/models/res_partner.py:222]: Remove

File: contact_modification/models/res_partner.py [hunk 3]
```diff
@@ -213,6 +214,27 @@ def _compute_task_count(self):
...
+        if not len(platform):
+            raise UserError("No Platform Connection Type existed.")
+        ctx = {"default_platform_type": platform[0].id}
+
+        return {
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "res_model": "platform.connector",
+            "target": "new",
+            # "res_id": self.task_id.id,
```
Comment (dmontooth) [inline contact_modification/models/res_partner.py:233]: Remove

File: contact_modification/views/res_partner_platform_connector.xml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                        </group>
+                    </group>                 
+                    <group colspan="1" col="1">
+                        <img style="margin:auto;display: block;width:100%;" src='/contact_modification/static/image/sub2account.png' attrs="{'invisible':[('connection_level', '!=', 'account')]}" alt="Connection Level"/>
+                        <img style="margin:auto;display: block;width:100%;" src='/contact_modification/static/image/sub2campaign.png' attrs="{'invisible':[('connection_level', '!=', 'campaign')]}" alt="Connection Level"/>
+                        <img style="margin:auto;display: block;width:100%;" src='/contact_modification/static/image/sub2adgroup.png' attrs="{'invisible':[('connection_level', '!=', 'adgroup')]}" alt="Connection Level"/>
+                    </group>           
+                </group>        
+
+                <script>
```
Comment (dmontooth) [inline contact_modification/views/res_partner_platform_connector.xml:92]: Looks like we can remove this whole script block.

**Review by dmontooth — CHANGES_REQUESTED** [review]
There are a handful of commented-out lines that can be removed. There is also a typo and verbiage change requested.

File: contact_modification/views/res_partner_platform_connector.xml (hunk 4 above)
Comment (Vlad-CL) [reply contact_modification/views/res_partner_platform_connector.xml:92]: sorry. 
this was experimental, and I forgot remove it it before publishing, sorry

File: contact_modification/models/platform_connection.py (hunk 1 above)
Comment (Vlad-CL) [reply contact_modification/models/platform_connection.py:87]: thank you

File: contact_modification/models/res_partner.py (hunk 3 above)
Comment (Vlad-CL) [reply contact_modification/models/res_partner.py:233]: thank you

**Review by dmontooth — APPROVED** [review]

**Pull request merged**